### ISSUE TITLE
Support for cleveref line numbers

### ISSUE THREAD

**Issue by dbogatov** [arnomi/cryptocode#5]
Hi,

Would it be possible to somehow add support for cleveref?
That is, if I have a labeled line number, I would like to say `On \cref{algorithm:name:line-2} we have` and observe "On line 2 we have" with a correct hyperlink.

Thank you!

**Comment by arnomi** [comment]
Hi,

I've included a `pcfixcleveref` command in version 0.43 (currently on develop). In the meantime, you can fix it locally as follows: Load cleveref after cryptocode and add the following after loading cleveref (still in the preamble)

```
\AtBeginDocument{
\pcfixhyperref
\makeatletter
\crefformat{@pclinenumber}{line~#2#1#3}
\crefrangeformat{@pclinenumber}{lines~#3#1#4 to~#5#2#6}
\makeatother
}
```

If you want to use hyperref in addition, load it before cryptocode. Following is a full example document:

```
\documentclass{article}

\usepackage{hyperref}
\usepackage{cryptocode}
\usepackage{cleveref}	

\AtBeginDocument{
\pcfixhyperref
\makeatletter
\crefformat{@pclinenumber}{line~#2#1#3}
\crefrangeformat{@pclinenumber}{lines~#3#1#4 to~#5#2#6}
\makeatother
}

\begin{document}

\pseudocodeblock[linenumbering]{
\label{ln:1}a\\
\label{ln:2}b\\
\label{ln:3}c\\
\label{ln:4}d}

See \crefrange{ln:1}{ln:4} in the above code example.

\end{document}
```

**Comment by dbogatov** [comment]
@arnomi , you are a magician!

The fix works perfectly.
I only have a tiny suggestion to include capitalized labels.

```latex
% https://github.com/arnomi/cryptocode/issues/5
\AtBeginDocument{
	\pcfixhyperref%
	\makeatletter
		\crefformat{@pclinenumber}{line~#2#1#3}
		\Crefformat{@pclinenumber}{Line~#2#1#3}
		\crefrangeformat{@pclinenumber}{lines~#3#1#4 to~#5#2#6}
		\Crefrangeformat{@pclinenumber}{Lines~#3#1#4 to~#5#2#6}
	\makeatother
}
```
